api/v1/Store.php

```diff
@@ -404,87 +429,105 @@ private function LoadJSON(string $jsonString, string $filename, string $dataStor
             );
         }
 
-        $layerId = -1;
-        $mspId = null;
-        $missingMspIds = 0;
-
         foreach ($data["features"] as $feature) {
-            if ($feature["geometry_name"] == "the_geom") {
-                $featureProperties = $feature["properties"];
-
-                $type = '0';
-                $countryId = 0;
-                $this->moveDataFromArray($featureProperties, $type, $mspId, $missingMspIds, $countryId);
-
-                $geometryData = $feature["geometry"];
-                if ($geometryData == null) {
-                    Log::LogWarning(
-                        "Could not import geometry with id ".$feature["id"]." of layer ".$filename.
-                        ". The feature in question has NULL geometry. Here's some property information to help find ".
-                        "the feature: MSP ID: ".$mspId." - ".substr(var_export($featureProperties, true), 0, 80)
-                    );
-                    continue;
-                }
+            $featureProperties = $feature["properties"];
+
+            $type = '0';
+            $mspId = null;
+            $countryId = null;
+            $this->moveDataFromArray($layerMetaData, $featureProperties, $type, $mspId, $countryId);
+
+            $geometryData = $feature["geometry"];
+            if ($geometryData == null) {
+                Log::LogWarning(
+                    "Could not import geometry with id ".$feature["id"]." of layer ".$filename.
+                    ". The feature in question has NULL geometry. Here's some property information to help find ".
+                    "the feature: MSP ID: ".$mspId." - ".substr(var_export($featureProperties, true), 0, 80)
+                );
+                continue;
+            }
+
+            $layerId = $this->getLayerId($filename, $geometryData["type"], $dataStore);
 
-                if ($layerId === -1) {
-                    // First geometry type defines layer type. Would like to pull this out but we need to find the first
-                    //   "the_geom" instance.
-                    $layerId = Database::GetInstance()->query(
-                        "INSERT INTO layer (layer_name, layer_geotype, layer_group) VALUES (?, ?, ?)",
-                        array($filename, $geometryData["type"], $dataStore),
-                        true
+            // let's make sure we are always working with multidata: multipolygon, multilinestring, multipoint
+            if ($geometryData["type"] == "Polygon"
+                || $geometryData["type"] == "LineString"
+                ||  $geometryData["type"] == "Point"
+            ) {
+                $geometryData["coordinates"] = [$geometryData["coordinates"]];
+                $geometryData["type"] = "Multi".$geometryData["type"];
+            }
+
+            $encodedFeatureData = json_encode($featureProperties);
+            if (strcasecmp($geometryData["type"], "MultiPolygon") == 0) {
```

Comment:
instead of this:
```
if (strcasecmp($geometryData["type"], "MultiPolygon") == 0) {
   // A
} elseif (strcasecmp($geometryData["type"], "MultiPoint") == 0) {
  // B
} elseif (strcasecmp($geometryData["type"], "MultiLineString") == 0) {
  // C
} else {
  // D
}
```

I would prefer:
```
if (strcasecmp($geometryData["type"], "MultiPolygon") == 0) {
   // A
   continue;
}
if (strcasecmp($geometryData["type"], "MultiPoint") == 0) {
   // B
   continue;
}
if (strcasecmp($geometryData["type"], "MultiLineString") == 0) {
   // C
   continue;
}
// D
```

At least the "D" code does not need extra indenting anymore. 
I always try to lessen the amount of indenting.

You could even move it to a seperate function "insertGeometryData" containg the above ifs ? of course the "continue" will become a return. 
```
foreach ($data["features"] as $feature) {
   // ... 
   $encodedFeatureData = json_encode($featureProperties);
   $this->insertGeometryData(...);
}
```

Note. Not mandatory stuff, see what you can do.

Reply:
I've done this slightly differently than you suggested. Have a look, hope this is good enough.

Reply:
fine
@hwarmelink you can resolve this conversation